app/controllers/courses_controller.rb

```diff
@@ -278,6 +280,29 @@ def update_courses_wikis
     @course.update_wikis(new_wikis)
   end
 
+  def course_wiki_namespaces_params
+    params.require(:course).permit(namespaces: [])[:namespaces]
+  end
+
+  def update_course_wiki_namespaces
+    namespaces = course_wiki_namespaces_params
```

Comment:
Adding a guard clause here (`return unless namespaces.present?` or similar) might be all that's needed to make this PR independent of frontend changes and able to be merged and deployed separately.

Reply:
Quite surprisingly for me, even without the guard clause, it worked just fine. The controller did not raise any error.
Should I still add a guard clause here?

Reply:
Okay....the clause is anyways necessary as some spec tests are failing...have added the guard clause now.